documentation/puppetdb_connection.markdown

```diff
@@ -0,0 +1,85 @@
+
+## Location
+
+The `puppetdb.conf` file is always located at `$confdir/puppetdb.conf`. Its location is **not** configurable.
+
+The location of the `confdir` varies; it depends on the OS, Puppet distribution, and user account. [See the confdir documentation for details.][confdir]
+
+## Example
+
+    [main]
+    server_urls = https://puppetdb.example.com:8081/
+
+## Format
+
+The `puppetdb.conf` file uses the same INI-like format as `puppet.conf`, but only uses a `[main]` section
+
+`[main]` Settings
+-----
+
+The `[main]` section defines all of the PuppetDB terminus settings.
+
+### `server_urls`
+
+This parameter specifies how the Puppet master should connect to PuppetDB. The configuration should look something like the configuration below:
+
+    server_urls = https://puppetdb.example.com:8081/
+
+Puppet _requires_ the use of PuppetDB's secure, HTTPS port. You cannot use the unencrypted, plain HTTP port. If you have specified a [`web-router-service` setting in your PuppetDB configuration][web_router_setting], that prefix must be reflected in your URLs. See [`url-prefix`][url_prefix_setting] for the (now deprecated) way of setting the `url-prefix`.
+
+Note that a comma separated list of the URLs can be provided if there are more than one PuppetDB instances available. A `server_urls` config that supports two PuppetDBs would look like:
+
+    server_urls = https://puppetdb1.example.com:8081/,https://puppetdb2.example.com:8081/
+
+The PuppetDB terminus will always attempt to connect to the first PuppetDB instance specified (listed above as puppetdb1). If a server-side exception occurs, or the request takes too long (see [`server_url_timeout`](#server_url_timeout)), the PuppetDB terminus will attempt the same operation on the next instance in the list.
+
+The default value is https://puppetdb:8081/
+
+### `server_url_timeout`
+
+The `server_url_timeout` field sets the maximum amount of time (in seconds) the PuppetDB termini will wait for an HTTP request to be responded to by PuppetDB. If the user has specified more than one PuppetDB URL and a timeout has occurred, it will attempt the same request on the next server in the list.
+
+The default value is 30 seconds.
+
+### `soft_write_failure`
+
+For availability reasons there is a setting named `soft_write_failure` that will cause the PuppetDB termini to fail in a soft-manner if PuppetDB is not accessable for command submission. This will mean that users who are either not using storeconfigs, or only exporting resources will still have their catalogs compile during a PuppetDB outage.
+
+The default value is false.
+
+### `server`
+
+> **Deprecated:** `server_urls` replaces this setting and will be removed in the future
```

Comment:
This is ambiguous. I'd edit it as:

    > **Deprecated:** `server_urls` replaces this setting. This setting will be removed in the future.

Also, what's the precedence of these settings? If `server_urls` is present, will `server` be ignored?